tests/datasource/data_connector/test_data_connector_query.py

```diff
@@ -494,13 +677,86 @@ def my_custom_batch_selector(batch_identifiers: dict) -> bool:
                 data_asset_name="TestFiles",
                 data_connector_query={
                     "custom_filter_function": my_custom_batch_selector,
-                    "index": (0, 4, 3),
+                    "index": "::-1",
                 },
```

Comment:
Thank you for this PR @alexsherstinsky. A small ask from me is some additional inline comments for the tests.  More specifically, if we could have them at two spots: 

1) where we define the `index` for each test (ie `"index": "::-1"`, like `starting at 0 index, stepping by 1, reverse order`)
2) and where we define the `expected` batch definition list (`all 5 batch definitions, in reverse order according to 'name' key`)

It would help if we were ever to come back to these tests in the future

Reply:
I did see that the description is the name of the tests, but this would be a small addition